app/src/main/java/woowacourse/movie/view/ext/IntentExt.kt

```
@@ -0,0 +1,16 @@
+package woowacourse.movie.view.ext
+
+import android.content.Intent
+import android.os.Build
+import java.io.Serializable
+
+inline fun <reified T : Serializable> Intent.getSerializable(
+    key: String,
+    clazz: Class<T>,
+): T {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        getSerializableExtra(key, clazz) ?: throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)